File: lib/py/uapi/internal/validation/GetTypeUnexpectedValidationFailure.py
from typing import List, Any, Dict

from uapi.internal.validation.ValidationFailure import ValidationFailure


def get_type_unexpected_validation_failure(path: List[Any], value: Any, expected_type: str) -> List['ValidationFailure']:
    from uapi.internal.types.GetType import get_type

    actual_type = get_type(value)
    data = {
        "actual": {actual_type: {}},
        "expected": {expected_type: {}}
    }
    return [
        ValidationFailure(path, "TypeUnexpected", data)
    ]

File: lib/ts/uapi/internal/validation/GetTypeUnexpectedValidationFailure.ts
import { ValidationFailure } from 'uapi/internal/validation/ValidationFailure';
import { getType } from 'uapi/internal/types/GetType';

export function getTypeUnexpectedValidationFailure(path: any[], value: any, expectedType: string): ValidationFailure[] {
    const actualType = getType(value);
    const data: { [key: string]: any } = {
        actual: { [actualType]: {} },
        expected: { [expectedType]: {} },
    };
    return [new ValidationFailure(path, 'TypeUnexpected', data)];
}
